src/game/state.ts

```diff
@@ -15,3 +15,24 @@ export const initGameState = (): GameState => ({
   turnNumber: 0,
   message: "Welcome to Kamon 🍱 ! Black player, you turn",
 });
+
+export const setGameAsDraw = (gameState: GameState): GameState => {
+  let newGameState = JSON.parse(JSON.stringify(gameState));
+
+  if (newGameState.winner != null) {
+    return;
+  }
+
+  newGameState.winner = null;
+  newGameState.isRunning = false;
+
+  console.log(" ");
+  console.log(" ");
+  console.log("==========================");
+  console.log(" ");
+  console.log("It's a draw !");
+  console.log(" ");
+  console.log("==========================");
+
+  return newGameState;
+};
```

Comment:
You are mixing game state updating logic with rendering logic. You should rather split the two.

Either add a boolean in `gameState` to store that the game is a draw, or create a helper fn that tells if a game is a draw or not from a given `gameState`.

Then, use this info in the rendering stage to display `It's a draw !`, and also end the game loop.

Reply:
Sames issue as https://github.com/marmelab/kamon/pull/13, where I split everything between rendering and game state computing

Reply:
I did separate everything though, my comment didn't make it obvious enough.